### PR TITLE
Add /scan route and scanning ability | slight routing initialize cleanup

### PR DESCRIPTION
**Pull request by cgmckeever**
Add /scan route and scanning ability 

Cleans the shared routing endpoint setup calls via private `baseRoutingSetup()`

### CODE REVIEW

File: src/ConfigManager.cpp [hunk 1]
```diff
@@ -303,6 +331,19 @@ void ConfigManager::startApi() {
     server->begin();
 }
 
+void ConfigManager::baseRoutingSetup() {
```
Comment (nrwiersma) [inline src/ConfigManager.cpp]: How about calling this `createBaseWebServer` to make it a bit clearer that it creates a new web server.

File: src/ConfigManager.cpp [hunk 2]
```diff
@@ -136,6 +136,48 @@ void ConfigManager::handleAPPost() {
...
+    if (n == 0) {
+        DebugPrintln("no networks found");
+    } else {
+        DebugPrint(n);
+        DebugPrintln(" networks found:");
+
+        for (int i = 0; i < n; ++i) {
+            String ssid = WiFi.SSID(i);
+            int rssi = WiFi.RSSI(i);
+            String security = WiFi.encryptionType(i) == ENC_TYPE_NONE ? "none" : "enabled";
```
Comment (nrwiersma) [inline src/ConfigManager.cpp]: Perhaps a bool here, to make it easier from a json perspective.

File: README.md [hunk 3]
```diff
@@ -200,13 +200,18 @@ server->on("/settings.html", HTTPMethod::HTTP_GET, [server](){
 
 # Endpoints
 
-### GET /
+
+## GET /
+
++ AP and API mode
```
Comment (nrwiersma) [inline README.md]: Aesthetic only: what about `###### *(AP,API)*` instead? Might be a little cleaner.

File: src/ConfigManager.cpp (hunk 2 above)
Comment (cgmckeever) [reply src/ConfigManager.cpp]: added via JSON L168

```
{
"ssid": "ESP_21557E",
"strength": -80,
"security": false
},
{
"ssid": "NETGEAR24",
"strength": -71,
"security": true
},


00:40:25.783 -> Name: ESP_21557E - Strength: -80 - Security: none
00:40:25.783 -> Name: NETGEAR24 - Strength: -71 - Security: enabled

```

File: README.md (hunk 3 above)
Comment (cgmckeever) [reply README.md]: pushed

**Pull request merged**